sympy/combinatorics/permutations.py

```diff
@@ -1569,6 +1542,10 @@ def __iter__(self):
         for i in self.array_form:
             yield i
 
+    def __repr__(self):
+        from sympy.printing.repr import srepr
+        return srepr(self)
```

Comment:
Why is this method needed? Does the repr-printer not default to the same output as srepr?

Reply:
`Basic.__repr__` uses `sstr` instead of `srepr`, so it had to be overrided manually.